terraformutils/providerwrapper/provider.go

```diff
@@ -237,14 +238,59 @@ func (p *ProviderWrapper) initProvider(verbose bool) error {
 }
 
 func getProviderFileName(providerName string) (string, error) {
+	providerFilePath, err := getProviderFileNameV13(providerName)
+	if err != nil || providerFilePath == "" {
+		return getProviderFileNameV12(providerName)
+	}
+	return providerFilePath, nil
+}
+
+func getProviderFileNameV13(providerName string) (string, error) {
+	defaultDataDir := os.Getenv("TF_DATA_DIR")
+	if defaultDataDir == "" {
+		defaultDataDir = DefaultDataDir
+	}
+	pluginPath := defaultDataDir + string(os.PathSeparator) + "plugins" + string(os.PathSeparator) +
+		"registry.terraform.io" + string(os.PathSeparator) + "hashicorp" + string(os.PathSeparator) + providerName
```

Comment:
I guess one comment is this hardcodes that the only supported providers will be 1st party hashicorp provided, I assume these are the only ones actually supported by terraformer for now, but still it might be nice to have some flexibility here for future use of other registries.  As per :  https://www.terraform.io/upgrade-guides/0-13.html#new-filesystem-layout-for-local-copies-of-providers .  Just a thought.

Reply:
good idea

We can add support this in the future, by provider name.
maybe save mapping providername=> registrie

Reply:
It must be fixed now, a lot of providers we support are custom ones, not officially recognized by Hashicorp.
That should be fine to support only `registry.terraform.io` registry.